### PR TITLE
Fix the bug that Secure Delegate Stubs are compiled every time.

### PR DESCRIPTION
**Pull request by sjsinju**
I found that Secure Delegate stubs are compiled every time during application execution, which has a negative impact on execution performance. #8554

Like the GetMulticastInvoke() method, GetSecureInvoke() checks the hashtable when the method is executed and uses it if it already exists.

### CODE REVIEW

**Comment by janvorli** [comment]
@sjsinju - could you please revert the changes to the file access rights that you've made to all the sources?
cc: @jkotas

**Comment by janvorli** [comment]
There is a build eror in the CI:
`error C2065: 'pDelMT': undeclared identifier`

**Comment by sjsinju** [comment]
@janvorli 
I fixed what you say and It seems that all checks are passed.

**Comment by sjsinju** [comment]
@janvorli @jkotas 
We need to merge this PR for our internal business.
If there is no problem, could you please merge this PR ASAP?

**Comment by jkotas** [comment]
LGTM. Thanks!

File: src/vm/comdelegate.cpp [hunk 1]
```diff
@@ -2931,47 +2931,61 @@ PCODE COMDelegate::GetSecureInvoke(MethodDesc* pMD)
...
+        // Load the "real" delegate
+        pCode->EmitLoadThis();
+        pCode->EmitLDFLD(pCode->GetToken(MscorlibBinder::GetField(FIELD__MULTICAST_DELEGATE__INVOCATION_LIST)));
 
-    SigTypeContext emptyContext;
-    ILStubLinker sl(pMD->GetModule(), pMD->GetSignature(), &emptyContext, pMD, TRUE, TRUE, FALSE);
+        // Load the arguments
+        UINT paramCount = 0;
+        while(paramCount < sig.NumFixedArgs())
+          pCode->EmitLDARG(paramCount++);
```
Comment (janvorli) [inline src/vm/comdelegate.cpp]: I've just noticed there are several indentation issues. Could you please create another PR to fix them?
The first one is here - each indentation should be 4 spaces, you have just two here.

File: src/vm/comdelegate.cpp [hunk 2]
```diff
@@ -2931,47 +2931,61 @@ PCODE COMDelegate::GetSecureInvoke(MethodDesc* pMD)
...
+        DWORD cbSig;
 
-    // Call the delegate
-    pCode->EmitCALL(pCode->GetToken(pMD), sig.NumFixedArgs(), fReturnVal);
+        pMD->GetSig(&pSig,&cbSig);
 
-    // Return
-    pCode->EmitRET();
+        MethodDesc* pStubMD =
+          ILStubCache::CreateAndLinkNewILStubMethodDesc(pMD->GetLoaderAllocator(),
```
Comment (janvorli) [inline src/vm/comdelegate.cpp]: Here is another one

**Comment by sjsinju** [comment]
@janvorli Sorry. I got it. I will create another PR to fix indentations you said.

**Pull request merged**